tests/acceptance/test_dbus.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import os
+import pytest
+import tempfile
+import time
+
+from fabric import Connection
+from multiprocessing import Process
+from paramiko.client import WarningPolicy
+
+from common import put_no_sftp
+
+MENDER_CONF = """{
+    "ClientProtocol": "https",
+    "HttpsClient": {
+        "SkipVerify": true
+    },
+    "InventoryPollIntervalSeconds": 5,
+    "RetryPollIntervalSeconds": 5,
+    "ServerCertificate": "/usr/share/doc/mender-client/examples/demo.crt",
+    "SkipVerify": true,
+    "ServerURL": "https://localhost:8443",
+    "DBus": {
+        "Enabled": true
+    }
+}
+"""
+
+
+@pytest.fixture
+def setup_mender_client_dbus(bitbake_variables, connection):
+    conffile = "/data/etc/mender/mender.conf"
+    backup = f"{conffile}.backup"
+    bdir = os.path.dirname(backup)
+    result = connection.run(
+        f"mkdir -p {bdir} && if [ -e {conffile} ]; then cp {conffile} {backup}; fi"
+    )
+    assert result.exited == 0
+
+    tf = tempfile.NamedTemporaryFile()
+    with open(tf.name, "w") as fd:
+        fd.write(MENDER_CONF)
+
+    put_no_sftp(tf.name, connection, remote=conffile)
+
+    try:
+        yield True
+    finally:
+        connection.run(
+            f"if [ -e {backup} ]; then dd if={backup} of=$(realpath {conffile}); fi"
+        )
```

Comment:
Being a fixture, you can also do:

```
@pytest.fixture
def setup_mender_client_dbus(request, bitbake_variables, connection):
    ...
    def fin():
        connection.run(...)
    request.addfinalizer(fin)
```

Both methods are fine, but as we use the `addfinalizer` one in integration tests, I wanted to point it out.

Reply:
Good idea to keep it consistent, refactored it to use the finalizer

Reply:
Yes, and there is one good reason to use a finalizer instead: IIRC, the yielding method doesn't work when pressing Ctrl-C to cancel the test, it only runs when the test succeeds. This may not be important in all cases, but for some it may be (cleaning up a huge test file, for example).